File: day01/julia/part2.jl
#!/usr/bin/env julia

using FromFile

@from "Helper.jl" import Helper

function main()
    # fname = "example.txt"
    fname = "input.txt"

    data = Helper.read_lines_as_ints(fname)
    sonar = zip(data, data[2:end], data[3:end])

    numbers = [sum(t) for t in sonar]
    # println(numbers)

    cnt = 0
    for i in 2:length(numbers)
        prev = numbers[i-1]
        curr = numbers[i]
        if curr > prev
            cnt += 1
        end
    end
    println(cnt)
end

if abspath(PROGRAM_FILE) == @__FILE__
    main()
end

File: day01/python/part2.py
#!/usr/bin/env python3

import helper

NOT_AVAILABLE = 0
NO_CHANGE = 0
INC = 1
DEC = -1

def main():
    # data = helper.read_lines_as_ints("example.txt")
    data = helper.read_lines_as_ints("input.txt")

    windows = zip(data, data[1:], data[2:])
    numbers = [sum(t) for t in windows]

    # print(numbers)

    sonar = [NOT_AVAILABLE]
    for i in range(1, len(numbers)):
        prev = numbers[i-1]
        curr = numbers[i]
        diff = curr - prev
        if diff > 0:
            sonar.append(INC)
        elif diff == 0:
            sonar.append(NO_CHANGE)
        else:
            sonar.append(DEC)
        #
    #
    result = sum([1 for value in sonar if value == INC])
    print(result)

##############################################################################

if __name__ == "__main__":
    main()
